### PR TITLE
Enable pre-commit hook for checking the copyright year

### PR DESCRIPTION
**Pull request by sleeepyjack**
Copied from cudf:
[copyright.py](https://github.com/rapidsai/cudf/blob/b14f1dcce526fae74b4b41b1a5e431bf493925ba/ci/checks/copyright.py)
[pre-commit hook](https://github.com/rapidsai/cudf/blob/b14f1dcce526fae74b4b41b1a5e431bf493925ba/.pre-commit-config.yaml#LL108C15-L113C53)

### CODE REVIEW

File: ci/checks/copyright.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+def modifiedFiles():
+    """Get a set of all modified files, as Diff objects.
+
+    The files returned have been modified in git since the merge base of HEAD
+    and the upstream of the target branch. We return the Diff objects so that
+    we can read only the staged changes.
+    """
+    repo = git.Repo()
+    # Use the environment variable TARGET_BRANCH or RAPIDS_BASE_BRANCH (defined in CI) if possible
+    target_branch = os.environ.get("TARGET_BRANCH", os.environ.get("RAPIDS_BASE_BRANCH"))
```
Comment (sleeepyjack) [inline ci/checks/copyright.py]: We probably need to define these variables in our CI too

File: ci/checks/copyright.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+import argparse
+import datetime
+import os
+import re
+import sys
+
+import git
+
+FilesToCheck = [
+    re.compile(r"[.](cmake|cpp|cu|cuh|h|hpp|sh|pxd|py|pyx)$"),
```
Comment (PointKernel) [inline ci/checks/copyright.py]:
```suggestion
    re.compile(r"[.](cmake|cpp|cu|cuh|h|hpp|inl|sh|py)$"),
```

File: ci/checks/copyright.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    re.compile(r"CMakeLists[.]txt$"),
+    re.compile(r"CMakeLists_standalone[.]txt$"),
+    re.compile(r"setup[.]cfg$"),
+    re.compile(r"[.]flake8[.]cython$"),
+    re.compile(r"meta[.]yaml$"),
```
Comment (PointKernel) [inline ci/checks/copyright.py]:
```suggestion
```
Can be moved since we don't have such files in cuco.

**Comment by sleeepyjack** [comment]
FYI The current state works locally but not in CI.

Dumb question: Does CI not clone+checkout the repo? Apparently the gitpython module isn't able to determine the correct branch with `repo.git.describe`.

**Comment by sleeepyjack** [comment]
@PointKernel do we know any RAPIDS folks who might be able to help with this?

**Comment by PointKernel** [comment]
> @PointKernel do we know any RAPIDS folks who might be able to help with this?

I wanted to reach out to @bdice for guidance but just got so occupied recently. Long story short, the cudf scripts work fine locally but not with `pre-commit.ci` e.g. [link](https://results.pre-commit.ci/run/github/226973898/1676077633.XpTxsfQiSSyFJak80bltOA). Any pointers on how to solve the issue would be highly appreciated.

**Comment by sleeepyjack** [comment]
Closing this PR as #279 proposes a better approach

**Pull request closed without merging**